Our first step is to install pandas in order to use it for a thorrow analysis of the dataset we have.

In [8]:
import pandas as pd

In [9]:
df = pd.read_csv(r'C:\Users\HP i5\Desktop\FSD\Python\kc_house_data.csv')

In [10]:
df

id             date     price  bedrooms  bathrooms  \
0      7129300520  20141013T000000  221900.0         3       1.00   
1      6414100192  20141209T000000  538000.0         3       2.25   
2      5631500400  20150225T000000  180000.0         2       1.00   
3      2487200875  20141209T000000  604000.0         4       3.00   
4      1954400510  20150218T000000  510000.0         3       2.00   
...           ...              ...       ...       ...        ...   
21608   263000018  20140521T000000  360000.0         3       2.50   
21609  6600060120  20150223T000000  400000.0         4       2.50   
21610  1523300141  20140623T000000  402101.0         2       0.75   
21611   291310100  20150116T000000  400000.0         3       2.50   
21612  1523300157  20141015T000000  325000.0         2       0.75   

       sqft_living  sqft_lot  floors  waterfront  view  ...  grade  \
0             1180      5650     1.0           0     0  ...      7   
1             2570      7242     2.0           0     0  ...      7   
2              770     10000     1.0           0     0  ...      6   
3             1960      5000     1.0           0     0  ...      7   
4             1680      8080     1.0           0     0  ...      8   
...            ...       ...     ...         ...   ...  ...    ...   
21608         1530      1131     3.0           0     0  ...      8   
21609         2310      5813     2.0           0     0  ...      8   
21610         1020      1350     2.0           0     0  ...      7   
21611         1600      2388     2.0           0     0  ...      8   
21612         1020      1076     2.0           0     0  ...      7   

       sqft_above  sqft_basement  yr_built  yr_renovated  zipcode      lat  \
0            1180              0      1955             0    98178  47.5112   
1            2170            400      1951          1991    98125  47.7210   
2             770              0      1933             0    98028  47.7379   
3            1050            910      1965             0    98136  47.5208   
4            1680              0      1987             0    98074  47.6168   
...           ...            ...       ...           ...      ...      ...   
21608        1530              0      2009             0    98103  47.6993   
21609        2310              0      2014             0    98146  47.5107   
21610        1020              0      2009             0    98144  47.5944   
21611        1600              0      2004             0    98027  47.5345   
21612        1020              0      2008             0    98144  47.5941   

          long  sqft_living15  sqft_lot15  
0     -122.257           1340        5650  
1     -122.319           1690        7639  
2     -122.233           2720        8062  
3     -122.393           1360        5000  
4     -122.045           1800        7503  
...        ...            ...         ...  
21608 -122.346           1530        1509  
21609 -122.362           1830        7200  
21610 -122.299           1020        2007  
21611 -122.069           1410        1287  
21612 -122.299           1020        1357  

[21613 rows x 21 columns]

In [11]:
df.shape

(21613, 21)

The df.shape allows to understand the size and structure of the dataframe at hand. 


In [12]:
df.columns

Index(['id', 'date', 'price', 'bedrooms', 'bathrooms', 'sqft_living',
       'sqft_lot', 'floors', 'waterfront', 'view', 'condition', 'grade',
       'sqft_above', 'sqft_basement', 'yr_built', 'yr_renovated', 'zipcode',
       'lat', 'long', 'sqft_living15', 'sqft_lot15'],
      dtype='object')

ID: Unique identifier for each appartment, date when the appartment data was collected, price:target variable, the result we are studying. and the other attributes are features of the appartment that influence the price


In [13]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 21613 entries, 0 to 21612
Data columns (total 21 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   id             21613 non-null  int64  
 1   date           21613 non-null  object 
 2   price          21613 non-null  float64
 3   bedrooms       21613 non-null  int64  
 4   bathrooms      21613 non-null  float64
 5   sqft_living    21613 non-null  int64  
 6   sqft_lot       21613 non-null  int64  
 7   floors         21613 non-null  float64
 8   waterfront     21613 non-null  int64  
 9   view           21613 non-null  int64  
 10  condition      21613 non-null  int64  
 11  grade          21613 non-null  int64  
 12  sqft_above     21613 non-null  int64  
 13  sqft_basement  21613 non-null  int64  
 14  yr_built       21613 non-null  int64  
 15  yr_renovated   21613 non-null  int64  
 16  zipcode        21613 non-null  int64  
 17  lat            21613 non-null  float64
 18  long  

The df.info method is very useful as it gives a summary of the pandas data frame, it give the number of columns (variables) and mentions each of them, it provides the number of non-null value count, in addition to the data types corresponding to each varibale (int, float, object). Moreover, it provides an estimate of the memory used by the data frame.

In [14]:
df.head()

id             date     price  bedrooms  bathrooms  sqft_living  \
0  7129300520  20141013T000000  221900.0         3       1.00         1180   
1  6414100192  20141209T000000  538000.0         3       2.25         2570   
2  5631500400  20150225T000000  180000.0         2       1.00          770   
3  2487200875  20141209T000000  604000.0         4       3.00         1960   
4  1954400510  20150218T000000  510000.0         3       2.00         1680   

   sqft_lot  floors  waterfront  view  ...  grade  sqft_above  sqft_basement  \
0      5650     1.0           0     0  ...      7        1180              0   
1      7242     2.0           0     0  ...      7        2170            400   
2     10000     1.0           0     0  ...      6         770              0   
3      5000     1.0           0     0  ...      7        1050            910   
4      8080     1.0           0     0  ...      8        1680              0   

   yr_built  yr_renovated  zipcode      lat     long  sqft_living15  \
0      1955             0    98178  47.5112 -122.257           1340   
1      1951          1991    98125  47.7210 -122.319           1690   
2      1933             0    98028  47.7379 -122.233           2720   
3      1965             0    98136  47.5208 -122.393           1360   
4      1987             0    98074  47.6168 -122.045           1800   

   sqft_lot15  
0        5650  
1        7639  
2        8062  
3        5000  
4        7503  

[5 rows x 21 columns]

In [15]:
df[df.waterfront > 0]

id             date      price  bedrooms  bathrooms  \
49      822039084  20150311T000000  1350000.0         3       2.50   
230    8096000060  20150413T000000   655000.0         2       1.75   
246    2025069065  20140929T000000  2400000.0         4       2.50   
264    2123039032  20141027T000000   369900.0         1       0.75   
300    3225069065  20140624T000000  3080000.0         4       5.00   
...           ...              ...        ...       ...        ...   
19984  2025069140  20150317T000000  1900000.0         3       2.50   
20325   518500480  20140811T000000  3000000.0         3       3.50   
20767  8043700105  20150417T000000  2300000.0         4       4.00   
21201   518500460  20141008T000000  2230000.0         3       3.50   
21576  9253900271  20150107T000000  3570000.0         5       4.50   

       sqft_living  sqft_lot  floors  waterfront  view  ...  grade  \
49            2753     65005     1.0           1     2  ...      9   
230           1450     15798     2.0           1     4  ...      7   
246           3650      8354     1.0           1     4  ...      9   
264            760     10079     1.0           1     4  ...      5   
300           4550     18641     1.0           1     4  ...     10   
...            ...       ...     ...         ...   ...  ...    ...   
19984         2830      4334     3.0           1     4  ...     10   
20325         4410     10756     2.0           1     4  ...     11   
20767         4360      8175     2.5           1     4  ...     10   
21201         3760      5634     2.0           1     4  ...     11   
21576         4850     10584     2.0           1     4  ...     10   

       sqft_above  sqft_basement  yr_built  yr_renovated  zipcode      lat  \
49           2165            588      1953             0    98070  47.4041   
230          1230            220      1915          1978    98166  47.4497   
246          1830           1820      2000             0    98074  47.6338   
264           760              0      1936             0    98070  47.4683   
300          2600           1950      2002             0    98074  47.6053   
...           ...            ...       ...           ...      ...      ...   
19984        2830              0      2006             0    98074  47.6318   
20325        3430            980      2014             0    98056  47.5283   
20767        3940            420      2007             0    98008  47.5724   
21201        2830            930      2014             0    98056  47.5285   
21576        3540           1310      2007             0    98008  47.5943   

          long  sqft_living15  sqft_lot15  
49    -122.451           2680       72513  
230   -122.375           2030       13193  
246   -122.072           3120       18841  
264   -122.438           1230       14267  
300   -122.077           4550       19508  
...        ...            ...         ...  
19984 -122.071           2830       38211  
20325 -122.205           3550        5634  
20767 -122.104           2670        8525  
21201 -122.205           3560        5762  
21576 -122.110           3470       18270  

[163 rows x 21 columns]

Tried to check the effect of the presence of the waterfront on the prices, however it doesnt have much influence, as we can see, not much appartments do have a waterfront (Only 163 out of 21613) and there isnt much impact on the price. 

In [16]:
df.nsmallest(5, 'floors')

id             date      price  bedrooms  bathrooms  sqft_living  \
0  7129300520  20141013T000000   221900.0         3        1.0         1180   
2  5631500400  20150225T000000   180000.0         2        1.0          770   
3  2487200875  20141209T000000   604000.0         4        3.0         1960   
4  1954400510  20150218T000000   510000.0         3        2.0         1680   
5  7237550310  20140512T000000  1230000.0         4        4.5         5420   

   sqft_lot  floors  waterfront  view  ...  grade  sqft_above  sqft_basement  \
0      5650     1.0           0     0  ...      7        1180              0   
2     10000     1.0           0     0  ...      6         770              0   
3      5000     1.0           0     0  ...      7        1050            910   
4      8080     1.0           0     0  ...      8        1680              0   
5    101930     1.0           0     0  ...     11        3890           1530   

   yr_built  yr_renovated  zipcode      lat     long  sqft_living15  \
0      1955             0    98178  47.5112 -122.257           1340   
2      1933             0    98028  47.7379 -122.233           2720   
3      1965             0    98136  47.5208 -122.393           1360   
4      1987             0    98074  47.6168 -122.045           1800   
5      2001             0    98053  47.6561 -122.005           4760   

   sqft_lot15  
0        5650  
2        8062  
3        5000  
4        7503  
5      101930  

[5 rows x 21 columns]

In [17]:
df.nlargest(5, 'floors')

id             date      price  bedrooms  bathrooms  \
875    6306400140  20140612T000000  1100000.0         0        0.0   
10077  1972202010  20140801T000000   435000.0         3        3.0   
11594  3180100023  20150130T000000   544000.0         3        2.5   
14885  8673400177  20150402T000000   525000.0         3        3.0   
15424  1702900664  20150416T000000   479000.0         2        2.5   

       sqft_living  sqft_lot  floors  waterfront  view  ...  grade  \
875           3064      4764     3.5           0     2  ...      7   
10077         1440      1350     3.5           0     2  ...      8   
11594         1760      1755     3.5           0     0  ...      8   
14885         1730      1074     3.5           0     0  ...      8   
15424         1730      1037     3.5           0     0  ...      8   

       sqft_above  sqft_basement  yr_built  yr_renovated  zipcode      lat  \
875          3064              0      1990             0    98102  47.6362   
10077        1440              0      2005             0    98103  47.6525   
11594        1760              0      1998             0    98105  47.6688   
14885        1730              0      2006             0    98107  47.6692   
15424        1730              0      2008             0    98118  47.5594   

          long  sqft_living15  sqft_lot15  
875   -122.322           2360        4000  
10077 -122.345           1440        1350  
11594 -122.279           1700        1721  
14885 -122.392           1370        1185  
15424 -122.285           1280        1026  

[5 rows x 21 columns]

By comparing the appartments with 3.5 floors with apartments with the minimum as 1 floor only, we can see that it doesn't have a real impact on the pricing, however it is clear that the sqft_living is the attribute that is affecting the pricing much.
As comparing app nb 4 to that nb 11594 of similar sqft_living, nb of bedrooms, nb of bathrooms, 0 waterfronts both, same view/grade.. but the first has only one floor and the second has 3.5 floor, we can see that there isnt much difference in the pricing, meaning that the increment of floors didn't impact the pricing. 

In [18]:
df.nlargest(10, 'sqft_living')

id             date      price  bedrooms  bathrooms  \
12777  1225069038  20140505T000000  2280000.0         7       8.00   
7252   6762700020  20141013T000000  7700000.0         6       8.00   
3914   9808700762  20140611T000000  7060000.0         5       4.50   
9254   9208900037  20140919T000000  6890000.0         6       7.75   
8092   1924059029  20140617T000000  4670000.0         5       6.75   
4411   2470100110  20140804T000000  5570000.0         5       5.75   
14556  2303900035  20140611T000000  2890000.0         5       6.25   
18302  6072800246  20140702T000000  3300000.0         5       6.25   
1164   1247600105  20141020T000000  5110000.0         5       5.25   
1448   8907500070  20150413T000000  5350000.0         5       5.00   

       sqft_living  sqft_lot  floors  waterfront  view  ...  grade  \
12777        13540    307752     3.0           0     4  ...     12   
7252         12050     27600     2.5           0     3  ...     13   
3914         10040     37325     2.0           1     2  ...     11   
9254          9890     31374     2.0           0     4  ...     13   
8092          9640     13068     1.0           1     4  ...     12   
4411          9200     35069     2.0           0     0  ...     13   
14556         8670     64033     2.0           0     4  ...     13   
18302         8020     21738     2.0           0     0  ...     11   
1164          8010     45517     2.0           1     4  ...     12   
1448          8000     23985     2.0           0     4  ...     12   

       sqft_above  sqft_basement  yr_built  yr_renovated  zipcode      lat  \
12777        9410           4130      1999             0    98053  47.6675   
7252         8570           3480      1910          1987    98102  47.6298   
3914         7680           2360      1940          2001    98004  47.6500   
9254         8860           1030      2001             0    98039  47.6305   
8092         4820           4820      1983          2009    98040  47.5570   
4411         6200           3000      2001             0    98039  47.6289   
14556        6120           2550      1965          2003    98177  47.7295   
18302        8020              0      2001             0    98006  47.5675   
1164         5990           2020      1999             0    98033  47.6767   
1448         6720           1280      2009             0    98004  47.6232   

          long  sqft_living15  sqft_lot15  
12777 -121.986           4850      217800  
7252  -122.323           3940        8800  
3914  -122.214           3930       25449  
9254  -122.240           4540       42730  
8092  -122.210           3270       10454  
4411  -122.233           3560       24345  
14556 -122.372           4140       81021  
18302 -122.189           4160       18969  
1164  -122.211           3430       26788  
1448  -122.220           4600       21750  

[10 rows x 21 columns]

In [19]:
df.nsmallest(10, 'sqft_living')

id             date     price  bedrooms  bathrooms  \
19452  3980300371  20140926T000000  142000.0         0       0.00   
15381  2856101479  20140701T000000  276000.0         1       0.75   
860    1723049033  20140620T000000  245000.0         1       0.75   
18379  1222029077  20141029T000000  265000.0         0       0.75   
4868   6896300380  20141002T000000  228000.0         0       1.00   
21332  9266700190  20150511T000000  245000.0         1       1.00   
8623   6303400395  20150130T000000  325000.0         1       0.75   
11500  4322200105  20150331T000000  229050.0         1       1.00   
14466  7549801385  20140612T000000  280000.0         1       0.75   
465    8658300340  20140523T000000   80000.0         1       0.75   

       sqft_living  sqft_lot  floors  waterfront  view  ...  grade  \
19452          290     20875     1.0           0     0  ...      1   
15381          370      1801     1.0           0     0  ...      5   
860            380     15000     1.0           0     0  ...      5   
18379          384    213444     1.0           0     0  ...      4   
4868           390      5900     1.0           0     0  ...      4   
21332          390      2000     1.0           0     0  ...      6   
8623           410      8636     1.0           0     0  ...      4   
11500          420      3298     1.0           0     0  ...      4   
14466          420      6720     1.0           0     0  ...      5   
465            430      5050     1.0           0     0  ...      4   

       sqft_above  sqft_basement  yr_built  yr_renovated  zipcode      lat  \
19452         290              0      1963             0    98024  47.5308   
15381         370              0      1923             0    98117  47.6778   
860           380              0      1963             0    98168  47.4810   
18379         384              0      2003             0    98070  47.4177   
4868          390              0      1953             0    98118  47.5260   
21332         390              0      1920             0    98103  47.6938   
8623          410              0      1953             0    98146  47.5077   
11500         420              0      1949             0    98136  47.5375   
14466         420              0      1922             0    98108  47.5520   
465           430              0      1912             0    98014  47.6499   

          long  sqft_living15  sqft_lot15  
19452 -121.888           1620       22850  
15381 -122.389           1340        5000  
860   -122.323           1170       15000  
18379 -122.491           1920      224341  
4868  -122.261           2170        6000  
21332 -122.347           1340        5100  
8623  -122.357           1190        8636  
11500 -122.391           1460        4975  
14466 -122.311           1420        6720  
465   -121.909           1200        7500  

[10 rows x 21 columns]

It is clear for sure that the sqft_living is highly affecting the pricing of the appartments, and that is approximately by a factor of 10. It also impacts the nb of bedrooms and bathrooms inside each apartment, as the square feet of living space increases the nb of bathrooms and bedrooms increase proportionally. This indicates that higher living areas result in higher prices.


In [45]:
df.nlargest(5, 'price')

id             date      price  bedrooms  bathrooms  \
7252  6762700020  20141013T000000  7700000.0         6       8.00   
3914  9808700762  20140611T000000  7060000.0         5       4.50   
9254  9208900037  20140919T000000  6890000.0         6       7.75   
4411  2470100110  20140804T000000  5570000.0         5       5.75   
1448  8907500070  20150413T000000  5350000.0         5       5.00   

      sqft_living  sqft_lot  floors  waterfront  view  ...  grade  sqft_above  \
7252        12050     27600     2.5           0     3  ...     13        8570   
3914        10040     37325     2.0           1     2  ...     11        7680   
9254         9890     31374     2.0           0     4  ...     13        8860   
4411         9200     35069     2.0           0     0  ...     13        6200   
1448         8000     23985     2.0           0     4  ...     12        6720   

      sqft_basement  yr_built  yr_renovated  zipcode      lat     long  \
7252           3480      1910          1987    98102  47.6298 -122.323   
3914           2360      1940          2001    98004  47.6500 -122.214   
9254           1030      2001             0    98039  47.6305 -122.240   
4411           3000      2001             0    98039  47.6289 -122.233   
1448           1280      2009             0    98004  47.6232 -122.220   

      sqft_living15  sqft_lot15  
7252           3940        8800  
3914           3930       25449  
9254           4540       42730  
4411           3560       24345  
1448           4600       21750  

[5 rows x 21 columns]

In [46]:
df.nsmallest(5, 'price')

id             date    price  bedrooms  bathrooms  sqft_living  \
1149   3421079032  20150217T000000  75000.0         1       0.00          670   
15293    40000362  20140506T000000  78000.0         2       1.00          780   
465    8658300340  20140523T000000  80000.0         1       0.75          430   
16198  3028200080  20150324T000000  81000.0         2       1.00          730   
8274   3883800011  20141105T000000  82000.0         3       1.00          860   

       sqft_lot  floors  waterfront  view  ...  grade  sqft_above  \
1149      43377     1.0           0     0  ...      3         670   
15293     16344     1.0           0     0  ...      5         780   
465        5050     1.0           0     0  ...      4         430   
16198      9975     1.0           0     0  ...      5         730   
8274      10426     1.0           0     0  ...      6         860   

       sqft_basement  yr_built  yr_renovated  zipcode      lat     long  \
1149               0      1966             0    98022  47.2638 -121.906   
15293              0      1942             0    98168  47.4739 -122.280   
465                0      1912             0    98014  47.6499 -121.909   
16198              0      1943             0    98168  47.4808 -122.315   
8274               0      1954             0    98146  47.4987 -122.341   

       sqft_living15  sqft_lot15  
1149            1160       42882  
15293           1700       10387  
465             1200        7500  
16198            860        9000  
8274            1140       11250  

[5 rows x 21 columns]

While comparing the highest and lowest prices, we may say that square feet of living space has the highest effect on the price on the appartment. With the increase of the square feet of living space, the number of bathrooms and bedrooms increase correspondingly. The grade the appartment is on also has an impact, higher grades have higher prices. Also, the sqaure feet of basement has an impact as well, where the appartments with 0 square feet of basement space are of the lowest prices.

In [21]:
with pd.option_context('display.max_rows', 15, 'display.max_columns', None):
    display(df)

id             date     price  bedrooms  bathrooms  \
0      7129300520  20141013T000000  221900.0         3       1.00   
1      6414100192  20141209T000000  538000.0         3       2.25   
2      5631500400  20150225T000000  180000.0         2       1.00   
3      2487200875  20141209T000000  604000.0         4       3.00   
4      1954400510  20150218T000000  510000.0         3       2.00   
...           ...              ...       ...       ...        ...   
21608   263000018  20140521T000000  360000.0         3       2.50   
21609  6600060120  20150223T000000  400000.0         4       2.50   
21610  1523300141  20140623T000000  402101.0         2       0.75   
21611   291310100  20150116T000000  400000.0         3       2.50   
21612  1523300157  20141015T000000  325000.0         2       0.75   

       sqft_living  sqft_lot  floors  waterfront  view  condition  grade  \
0             1180      5650     1.0           0     0          3      7   
1             2570      7242     2.0           0     0          3      7   
2              770     10000     1.0           0     0          3      6   
3             1960      5000     1.0           0     0          5      7   
4             1680      8080     1.0           0     0          3      8   
...            ...       ...     ...         ...   ...        ...    ...   
21608         1530      1131     3.0           0     0          3      8   
21609         2310      5813     2.0           0     0          3      8   
21610         1020      1350     2.0           0     0          3      7   
21611         1600      2388     2.0           0     0          3      8   
21612         1020      1076     2.0           0     0          3      7   

       sqft_above  sqft_basement  yr_built  yr_renovated  zipcode      lat  \
0            1180              0      1955             0    98178  47.5112   
1            2170            400      1951          1991    98125  47.7210   
2             770              0      1933             0    98028  47.7379   
3            1050            910      1965             0    98136  47.5208   
4            1680              0      1987             0    98074  47.6168   
...           ...            ...       ...           ...      ...      ...   
21608        1530              0      2009             0    98103  47.6993   
21609        2310              0      2014             0    98146  47.5107   
21610        1020              0      2009             0    98144  47.5944   
21611        1600              0      2004             0    98027  47.5345   
21612        1020              0      2008             0    98144  47.5941   

          long  sqft_living15  sqft_lot15  
0     -122.257           1340        5650  
1     -122.319           1690        7639  
2     -122.233           2720        8062  
3     -122.393           1360        5000  
4     -122.045           1800        7503  
...        ...            ...         ...  
21608 -122.346           1530        1509  
21609 -122.362           1830        7200  
21610 -122.299           1020        2007  
21611 -122.069           1410        1287  
21612 -122.299           1020        1357  

[21613 rows x 21 columns]

In [22]:
df.nlargest(3, "condition")

id             date     price  bedrooms  bathrooms  sqft_living  \
3   2487200875  20141209T000000  604000.0         4        3.0         1960   
25  1202000200  20141103T000000  233000.0         3        2.0         1710   
27  3303700376  20141201T000000  667000.0         3        1.0         1400   

    sqft_lot  floors  waterfront  view  ...  grade  sqft_above  sqft_basement  \
3       5000     1.0           0     0  ...      7        1050            910   
25      4697     1.5           0     0  ...      6        1710              0   
27      1581     1.5           0     0  ...      8        1400              0   

    yr_built  yr_renovated  zipcode      lat     long  sqft_living15  \
3       1965             0    98136  47.5208 -122.393           1360   
25      1941             0    98002  47.3048 -122.218           1030   
27      1909             0    98112  47.6221 -122.314           1860   

    sqft_lot15  
3         5000  
25        4705  
27        3861  

[3 rows x 21 columns]

In [23]:
df.nsmallest(3, "condition")

id             date     price  bedrooms  bathrooms  sqft_living  \
36   9435300030  20140528T000000  550000.0         4        1.0         1660   
380  1726059053  20140916T000000  270000.0         2        1.5         1380   
397  5175800060  20140623T000000  365000.0         4        2.0         1940   

     sqft_lot  floors  waterfront  view  ...  grade  sqft_above  \
36      34848     1.0           0     0  ...      5         930   
380    209959     1.0           0     0  ...      6        1380   
397     25600     1.0           0     0  ...      8        1940   

     sqft_basement  yr_built  yr_renovated  zipcode      lat     long  \
36             730      1933             0    98052  47.6621 -122.132   
380              0      1954             0    98011  47.7461 -122.195   
397              0      1962             0    98006  47.5722 -122.129   

     sqft_living15  sqft_lot15  
36            2160       11467  
380           3130       19868  
397           2000       10071  

[3 rows x 21 columns]

In [24]:
df['condition'].value_counts()

3    14031
4     5679
5     1701
2      172
1       30
Name: condition, dtype: int64

In [25]:
df[df.condition == 5] 

id             date      price  bedrooms  bathrooms  \
3      2487200875  20141209T000000   604000.0         4       3.00   
25     1202000200  20141103T000000   233000.0         3       2.00   
27     3303700376  20141201T000000   667000.0         3       1.00   
49      822039084  20150311T000000  1350000.0         3       2.50   
62     5072410070  20141021T000000   505000.0         3       1.75   
...           ...              ...        ...       ...        ...   
19604  7518504130  20140626T000000   663000.0         3       2.00   
19620  3902100175  20140728T000000   850000.0         5       3.00   
19873  3449000060  20141001T000000   320000.0         3       1.00   
19905  2888000030  20140926T000000   500000.0         4       2.25   
21303   522059130  20150429T000000   465000.0         3       1.00   

       sqft_living  sqft_lot  floors  waterfront  view  ...  grade  \
3             1960      5000     1.0           0     0  ...      7   
25            1710      4697     1.5           0     0  ...      6   
27            1400      1581     1.5           0     0  ...      8   
49            2753     65005     1.0           1     2  ...      9   
62            2519      8690     2.0           0     0  ...      8   
...            ...       ...     ...         ...   ...  ...    ...   
19604         1480      3876     1.0           0     0  ...      7   
19620         3900      5250     1.5           0     1  ...      8   
19873         1400      9000     1.0           0     0  ...      7   
19905         2270      8196     1.0           0     0  ...      7   
21303         1150     18200     1.0           0     0  ...      7   

       sqft_above  sqft_basement  yr_built  yr_renovated  zipcode      lat  \
3            1050            910      1965             0    98136  47.5208   
25           1710              0      1941             0    98002  47.3048   
27           1400              0      1909             0    98112  47.6221   
49           2165            588      1953             0    98070  47.4041   
62           2519              0      1973             0    98166  47.4428   
...           ...            ...       ...           ...      ...      ...   
19604         860            620      1928             0    98117  47.6808   
19620        2620           1280      1931             0    98116  47.5577   
19873        1400              0      1959             0    98059  47.5022   
19905        1150           1120      1963             0    98034  47.7214   
21303        1150              0      1959             0    98058  47.4262   

          long  sqft_living15  sqft_lot15  
3     -122.393           1360        5000  
25    -122.218           1030        4705  
27    -122.314           1860        3861  
49    -122.451           2680       72513  
62    -122.344           2500        9500  
...        ...            ...         ...  
19604 -122.382           1660        3774  
19620 -122.389           1950        5700  
19873 -122.145           1440        8400  
19905 -122.227           1920       10122  
21303 -122.187           1714       18200  

[1701 rows x 21 columns]

In [26]:
df[df.condition == 1] 

id             date      price  bedrooms  bathrooms  \
36     9435300030  20140528T000000   550000.0         4       1.00   
380    1726059053  20140916T000000   270000.0         2       1.50   
397    5175800060  20140623T000000   365000.0         4       2.00   
1442   5694000710  20141107T000000   352950.0         3       1.00   
1734    913000340  20150102T000000   252000.0         1       1.00   
2223   3886902445  20150316T000000   535000.0         2       1.00   
3004   9187200245  20141231T000000   441000.0         4       1.50   
3202    723049596  20140509T000000   255000.0         2       1.00   
3975   6324000090  20150511T000000   210000.0         2       1.00   
4651    859000110  20141002T000000   125000.0         1       1.00   
7376   9275200080  20141107T000000   295000.0         3       1.50   
7636   4008400515  20150120T000000   190000.0         1       0.75   
12306  2570600140  20150128T000000   196000.0         3       2.25   
12453  4022900951  20150402T000000   305000.0         2       1.00   
12668  4006000281  20140729T000000   227000.0         3       1.75   
13628  2023049218  20140716T000000   105500.0         2       1.00   
13629  2023049218  20150316T000000   445000.0         2       1.00   
14987  9187200095  20141202T000000   432500.0         6       2.00   
15293    40000362  20140506T000000    78000.0         2       1.00   
15337  7849200635  20140630T000000   235000.0         2       1.00   
15371  7936500221  20150114T000000   658000.0         2       1.00   
15712  3089000005  20140724T000000   150000.0         2       1.00   
16198  3028200080  20150324T000000    81000.0         2       1.00   
16893  3501100050  20141210T000000   125000.0         3       1.00   
16942  5015001680  20140611T000000   427000.0         4       1.00   
17805  2206900065  20150501T000000   380000.0         3       1.50   
18332  9272202260  20140924T000000   130000.0         3       1.00   
18645  7942600910  20141216T000000   575000.0         1       1.00   
18876  9808700025  20150211T000000  1500000.0         3       1.50   
19452  3980300371  20140926T000000   142000.0         0       0.00   

       sqft_living  sqft_lot  floors  waterfront  view  ...  grade  \
36            1660     34848     1.0           0     0  ...      5   
380           1380    209959     1.0           0     0  ...      6   
397           1940     25600     1.0           0     0  ...      8   
1442          1760      3000     1.5           0     0  ...      6   
1734           680      1638     1.0           0     4  ...      6   
2223           920      9000     1.0           0     0  ...      6   
3004          1100      3300     1.0           0     0  ...      7   
3202           810      7980     1.0           0     0  ...      6   
3975           990      8140     1.0           0     0  ...      6   
4651           500      7440     1.0           0     0  ...      5   
7376           720      7450     1.0           0     1  ...      5   
7636           780     77603     1.0           0     0  ...      5   
12306         1510      9600     1.0           0     0  ...      7   
12453          910     22725     1.0           0     0  ...      6   
12668         2380     12681     1.0           0     0  ...      6   
13628          930      7740     1.0           0     0  ...      5   
13629          930      7740     1.0           0     0  ...      5   
14987         3080      5500     2.0           0     0  ...      7   
15293          780     16344     1.0           0     0  ...      5   
15337          900     28800     1.0           0     0  ...      6   
15371         1010     14244     1.0           1     4  ...      5   
15712          850     54000     1.5           0     0  ...      4   
16198          730      9975     1.0           0     0  ...      5   
16893         1230      4800     1.5           0     0  ...      6   
16942         1860      4736     1.5           0     0  ...      7   
17805         1430     1

In [27]:
N=2
df.head(N)

id             date     price  bedrooms  bathrooms  sqft_living  \
0  7129300520  20141013T000000  221900.0         3       1.00         1180   
1  6414100192  20141209T000000  538000.0         3       2.25         2570   

   sqft_lot  floors  waterfront  view  ...  grade  sqft_above  sqft_basement  \
0      5650     1.0           0     0  ...      7        1180              0   
1      7242     2.0           0     0  ...      7        2170            400   

   yr_built  yr_renovated  zipcode      lat     long  sqft_living15  \
0      1955             0    98178  47.5112 -122.257           1340   
1      1951          1991    98125  47.7210 -122.319           1690   

   sqft_lot15  
0        5650  
1        7639  

[2 rows x 21 columns]

In [28]:
df.bedrooms

0        3
1        3
2        2
3        4
4        3
        ..
21608    3
21609    4
21610    2
21611    3
21612    2
Name: bedrooms, Length: 21613, dtype: int64

In [29]:
df[['bedrooms']] #the double brackets give a more organized form, if only one bracket its gnna be less organized. 

bedrooms
0             3
1             3
2             2
3             4
4             3
...         ...
21608         3
21609         4
21610         2
21611         3
21612         2

[21613 rows x 1 columns]

In [30]:
start_index = 3
end_index = 8
df.bedrooms[start_index:end_index+1]

3    4
4    3
5    4
6    3
7    3
8    3
Name: bedrooms, dtype: int64

In [31]:
df[df.bedrooms>10]

id             date     price  bedrooms  bathrooms  \
8757   1773100755  20140821T000000  520000.0        11       3.00   
15870  2402100895  20140625T000000  640000.0        33       1.75   

       sqft_living  sqft_lot  floors  waterfront  view  ...  grade  \
8757          3000      4960     2.0           0     0  ...      7   
15870         1620      6000     1.0           0     0  ...      7   

       sqft_above  sqft_basement  yr_built  yr_renovated  zipcode      lat  \
8757         2400            600      1918          1999    98106  47.5560   
15870        1040            580      1947             0    98103  47.6878   

          long  sqft_living15  sqft_lot15  
8757  -122.363           1420        4960  
15870 -122.331           1330        4700  

[2 rows x 21 columns]

In [32]:
df[['bedrooms', 'price']] #we open and close the brackets twice bc we're specifying a list

bedrooms     price
0             3  221900.0
1             3  538000.0
2             2  180000.0
3             4  604000.0
4             3  510000.0
...         ...       ...
21608         3  360000.0
21609         4  400000.0
21610         2  402101.0
21611         3  400000.0
21612         2  325000.0

[21613 rows x 2 columns]

In [33]:
my_list = ['bedrooms', 'price', 'floors', 'waterfront'] #you have to be exact while wrting bc theyre case sensitive
df[my_list]

bedrooms     price  floors  waterfront
0             3  221900.0     1.0           0
1             3  538000.0     2.0           0
2             2  180000.0     1.0           0
3             4  604000.0     1.0           0
4             3  510000.0     1.0           0
...         ...       ...     ...         ...
21608         3  360000.0     3.0           0
21609         4  400000.0     2.0           0
21610         2  402101.0     2.0           0
21611         3  400000.0     2.0           0
21612         2  325000.0     2.0           0

[21613 rows x 4 columns]

In [34]:
bedrooms = df['bedrooms']
bedrooms

0        3
1        3
2        2
3        4
4        3
        ..
21608    3
21609    4
21610    2
21611    3
21612    2
Name: bedrooms, Length: 21613, dtype: int64

In [35]:
df['bedrooms'].value_counts() #reoccurance of each nb of bedrooms

3     9824
4     6882
2     2760
5     1601
6      272
1      199
7       38
0       13
8       13
9        6
10       3
11       1
33       1
Name: bedrooms, dtype: int64

In [36]:
df[df.bedrooms == 33]

id             date     price  bedrooms  bathrooms  \
15870  2402100895  20140625T000000  640000.0        33       1.75   

       sqft_living  sqft_lot  floors  waterfront  view  ...  grade  \
15870         1620      6000     1.0           0     0  ...      7   

       sqft_above  sqft_basement  yr_built  yr_renovated  zipcode      lat  \
15870        1040            580      1947             0    98103  47.6878   

          long  sqft_living15  sqft_lot15  
15870 -122.331           1330        4700  

[1 rows x 21 columns]

This number is extremely big to be in a single appartment (33 bedrooms in one apartment), so we had to check for the whole data concerning this appartment to check if it is a typo, and it is clear it is. This apartment is only 1620 sqft_living, hence not that big of living space, thus could never contain 33 bedrooms. So most obviously it is a typo.

In [37]:
df['price'].value_counts() #this might not be useful as we have a large range of value

450000.0    172
350000.0    172
550000.0    159
500000.0    152
425000.0    150
           ... 
278800.0      1
439888.0      1
354901.0      1
942000.0      1
402101.0      1
Name: price, Length: 3625, dtype: int64

In [38]:
df['bedrooms'].unique()

array([ 3,  2,  4,  5,  1,  6,  7,  0,  8,  9, 11, 10, 33], dtype=int64)

In [39]:
df['price'].max()

7700000.0

In [40]:
df['price'].min()

75000.0

In [41]:
df['bedrooms'].sum()

72854

In [42]:
df.nlargest(2, 'bedrooms')

id             date     price  bedrooms  bathrooms  \
15870  2402100895  20140625T000000  640000.0        33       1.75   
8757   1773100755  20140821T000000  520000.0        11       3.00   

       sqft_living  sqft_lot  floors  waterfront  view  ...  grade  \
15870         1620      6000     1.0           0     0  ...      7   
8757          3000      4960     2.0           0     0  ...      7   

       sqft_above  sqft_basement  yr_built  yr_renovated  zipcode      lat  \
15870        1040            580      1947             0    98103  47.6878   
8757         2400            600      1918          1999    98106  47.5560   

          long  sqft_living15  sqft_lot15  
15870 -122.331           1330        4700  
8757  -122.363           1420        4960  

[2 rows x 21 columns]

In [43]:
df.nsmallest(2, 'bedrooms')

id             date      price  bedrooms  bathrooms  \
875   6306400140  20140612T000000  1100000.0         0        0.0   
3119  3918400017  20150205T000000   380000.0         0        0.0   

      sqft_living  sqft_lot  floors  waterfront  view  ...  grade  sqft_above  \
875          3064      4764     3.5           0     2  ...      7        3064   
3119         1470       979     3.0           0     2  ...      8        1470   

      sqft_basement  yr_built  yr_renovated  zipcode      lat     long  \
875               0      1990             0    98102  47.6362 -122.322   
3119              0      2006             0    98133  47.7145 -122.356   

      sqft_living15  sqft_lot15  
875            2360        4000  
3119           1470        1399  

[2 rows x 21 columns]

In [44]:
df['price'].mean()

540182.1587933188